plugins/concourse-resource.rb

```diff
@@ -0,0 +1,80 @@
+require 'yaml'
+require 'json'
+require 'open3'
+
+##################################################################
+# Same thing as the resource_types property in your pipeline.yaml
+##################################################################
+
+$concourseResourceTypes = Hash.new
+class ConcourseResourceType
+  attr_reader :definition
+
+  def self.register(definition)
+    resourceType = ConcourseResourceType.new(definition)
+    name = resourceType.definition["name"]
+    $concourseResourceTypes[name] = resourceType
+  end
+
+  private
+  def initialize(definition)
+    @definition = YAML.load(definition)
+    # TODO: validate required properties
+  end
+end
+
+
+##################################################################
+# Same thing as the resources property in your pipeline.yaml
+##################################################################
+
+class ConcourseResourceDefinition
+  attr_reader :definition
+
+  def initialize(definition)
+    @definition = YAML.load(definition)
+    # TODO: validate required properties
+  end
+end
+
+
+##################################################################
+# Concourse build-in Resource Types
+##################################################################
+
+ConcourseResourceType.register("
+  name: github-release
+  type: docker-image
+  source:
+    repository: concourse/github-release-resource
+    tag: latest
+")
+
+
+##################################################################
+# Expose Function to rmk
+##################################################################
+
+module ConcourseResource
+
+    include Rmk::Tools
+  
+    def concourse_put(name, resource, put_parameters)
+      resourceType = $concourseResourceTypes[resource.definition["type"]]
```

Comment:
Put as much computation stuff as possible into the `job`block. This block is only executed, if a rebuild is required.